fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -0,0 +1,359 @@
+        return FDBRecordStore.newBuilder()
+                .setFormatVersion(FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION) // set to max to test newest features (unsafe for real deployments)
+                .setKeySpacePath(path)
+                .setContext(context)
+                .setMetaDataProvider(metaData);
+    }
+
+    private TestRecordsTextProto.SimpleDocument createSimpleDocument(long docId, String text, int group) {
+        return TestRecordsTextProto.SimpleDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setGroup(group)
+                .build();
+    }
+
+    private TestRecordsTextProto.ComplexDocument createComplexDocument(long docId, String text, String text2, int group) {
+        return TestRecordsTextProto.ComplexDocument.newBuilder()
+                .setDocId(docId)
+                .setText(text)
+                .setText2(text2)
+                .setGroup(group)
+                .build();
+    }
+
+
+    @Test
+    public void simpleInsertAndSearch() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, metaDataBuilder -> {
+                metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+                metaDataBuilder.addIndex(SIMPLE_DOC, SIMPLE_TEXT_SUFFIXES);
+            });
+            recordStore.saveRecord(createSimpleDocument(1623L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            RecordCursor<IndexEntry> indexEntries = recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, BY_LUCENE_QUERY, TupleRange.allOf(Tuple.from("idiot")), null, ScanProperties.FORWARD_SCAN);
+            assertEquals(1, recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, BY_LUCENE_QUERY,
+                    TupleRange.allOf(Tuple.from("idiot")), null, ScanProperties.FORWARD_SCAN)
+                    .getCount().join());
+            assertEquals(1, context.getTimer().getCounter(FDBStoreTimer.Counts.LOAD_SCAN_ENTRY).getCount());
+        }
+    }
+
+    @Test
+    public void testContinuation() {
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, metaDataBuilder -> {
+                metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+                metaDataBuilder.addIndex(SIMPLE_DOC, SIMPLE_TEXT_SUFFIXES);
+            });
+            recordStore.saveRecord(createSimpleDocument(1623L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1625L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1626L, DYLAN, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            assertEquals(2, recordStore.scanIndex(SIMPLE_TEXT_SUFFIXES, BY_LUCENE_QUERY,
+                    TupleRange.allOf(Tuple.from("idiot")), Ints.toByteArray(2), ScanProperties.FORWARD_SCAN)
```

Comment:
This manually sets the continuation to 2, I think it would be better to have it query look at 2 records, and then query again with the continuation, and make sure that you see all the appropriate entries.